src/prefect/serialization/state.py

```diff
@@ -32,14 +32,17 @@ class BaseStateSchema(ObjectSchema):
     class Meta:
         object_class = state.State
 
+    context = fields.Dict(key=fields.Str(), values=JSONCompatible(), allow_none=True)
     message = fields.String(allow_none=True)
     _result = Nested(StateResultSchema, allow_none=False, value_selection_fn=get_safe)
 
     @post_load
     def create_object(self, data: dict, **kwargs: Any) -> state.State:
         result_obj = data.pop("_result", result.NoResult)
+        context = data.pop("context", dict())
```

Comment:
We actually have a good pattern for this (and the above line, which I didn't notice until now).

If you adjust this serializer's `Meta`, it'll automatically exclude attributes from `__init__`. Then you can still assign them as in line 45.

```
class MySerializer:
    class Meta:
        object_class = WhatEver
		exclude_fields = ['context']
```

Reply:
Hm whenever I do that, I still have to `pop` context out of the data dictionary, or else it an error is raised.  Here's what I updated:

```python
class BaseStateSchema(ObjectSchema):    
    class Meta:                                                                                                                                                            
        object_class = state.State                                                                                                                                        
        exclude_fields = ["context"]                                                                                                                                      
                                                                                                                                                                          
    context = fields.Dict(key=fields.Str(), values=JSONCompatible(), allow_none=True)                                                                                     
    message = fields.String(allow_none=True)                                                                                                                              
    _result = Nested(StateResultSchema, allow_none=False, value_selection_fn=get_safe)                                                                                    
                                                                                                                                                                          
    @post_load                                                                                                                                                            
    def create_object(self, data: dict, **kwargs: Any) -> state.State:                                                                                                    
        result_obj = data.pop("_result", result.NoResult)                                                                                                                 
        data["result"] = result_obj                                                                                                                                       
        base_obj = super().create_object(data)                                                                                                                            
        base_obj.context = data.get("context", dict())                                                                                                                    
        return base_obj   
```